### DISCUSSION TITLE
How to setup environment variables in Vercel?

### DISCUSSION THREAD

**Discussion by TaqiRaza512** [vercel/next.js#62502] [Help]
### Summary

A developer of MERN stack, who wants to deploy his app on vercel platform and confused about the setting of environment variables. The problem I am facing is that I have built an app on MERN stack and it is working fine on local host. 
![Screenshot from 2024-02-25 15-45-43](https://github.com/vercel/next.js/assets/122045573/30d69228-8502-4ad8-9531-46af0ea4f05a)

I push the project to the github by ignoring the .env file. I go to the vercel platform, and added the environment variables manually in its platform.

My question is that is vercel is enough intelligent, that it will replace the environment variables within the code as well.
![image](https://github.com/vercel/next.js/assets/122045573/8c57ac05-24b2-4745-9573-66ea07723c7e)

Anyone who have enough knowledge about it please reply.

**Comment by icyJoseph** [comment]
Hi

You just need to add them to the dashboard. https://vercel.com/docs/projects/environment-variables

Then when your project is built, your variables per environment are loaded, so that the project can read them from `process.env`.

Does that answer your question? 

<img width="851" alt="Screenshot 2024-02-25 at 15 41 44" src="https://github.com/vercel/next.js/assets/21013447/38b26da4-4a43-4c53-969f-6b3248b0c11a">
